lib/bsdlib/nrf91_sockets.c

```diff
@@ -554,21 +564,42 @@ static int nrf91_socket_offload_getsockopt(int sd, int level, int optname,
 	int retval;
 	int nrf_level;
 	int nrf_optname;
+	struct nrf_timeval nrf_rcvtimeo = {0, 0};
+	void *nrf_optval = optval;
+	nrf_socklen_t nrf_optlen = (nrf_socklen_t)*optlen;
 
 	if (z_to_nrf_level(level, &nrf_level) < 0)
 		goto error;
 	if (z_to_nrf_optname(level, optname, &nrf_optname) < 0)
 		goto error;
 
-	retval = nrf_getsockopt(sd, nrf_level, nrf_optname, optval,
-				(nrf_socklen_t *)optlen);
+	if ((level == SOL_SOCKET) && (optname == SO_RCVTIMEO)) {
+		nrf_optval = &nrf_rcvtimeo;
+		nrf_optlen = sizeof(struct nrf_timeval);
+	}
 
-	if ((retval == 0) && optval &&
-	    (level == SOL_SOCKET) && (optname == SO_ERROR)) {
-		/* Use bsd_os_errno_set() to translate from nRF error
-		 * to native error. */
-		bsd_os_errno_set(*(int *)optval);
-		*(int *)optval = errno;
+	retval = nrf_getsockopt(sd, nrf_level, nrf_optname, nrf_optval,
+				&nrf_optlen);
+
+
+	if ((retval == 0) && optval) {
```

Comment:
Just a small observation.
Other places in this code, pointers are checked against `NULL`, as example, line 626 and 662.

So I believe it is clearer to the reader that this is indeed a pointer test by using
```suggestion
	if ((retval == 0) && (optval != NULL) {
```

Reply:
Thanks, @tejlmand. Changed and amended.